src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -589,6 +597,87 @@ class RNNOpRewriteLayoutPattern : public OpRewritePattern<ONNXOp> {
   }
 };
 
+// =============================================================================
+// Rewrite pattern for Power
+// =============================================================================
+
+class PowToMulRewritePattern : public OpRewritePattern<ONNXPowOp> {
+public:
+  using OpRewritePattern<ONNXPowOp>::OpRewritePattern;
+
+  PowToMulRewritePattern(MLIRContext *context, int64_t maxPower)
+      : OpRewritePattern(context), maxPower(maxPower) {}
+
+  LogicalResult matchAndRewrite(
+      ONNXPowOp powOp, PatternRewriter &rewriter) const override {
+    Operation *op = powOp.getOperation();
+    Location loc = powOp.getLoc();
+    int64_t exponent;
+    // Test legality
+    if (!CanExpandPowOpToMul(powOp, exponent))
+      return failure();
+
+    // Rewrite
+    MultiDialectBuilder<OnnxBuilder> create(rewriter, loc);
+    Value input = powOp.getX();
+
+    Value result;
+    Type resultType = powOp.getZ().getType();
+    Type elementType = getElementTypeOrSelf(resultType);
+    if (exponent == 0) {
+      DenseElementsAttr valAttr;
+      if (elementType.isa<FloatType>())
+        valAttr = DenseElementsAttr::get(resultType, ArrayRef<float>({1.0}));
+      else if (elementType.isa<IntegerType>())
+        valAttr = DenseElementsAttr::get(resultType, ArrayRef<int64_t>({1}));
```

Comment:
I believe this will only work if elementType is F32 or I64, see https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/BuiltinAttributes.h#L108-L120

I suggest you create a 1 attribute similar to `getIdentity()` in ConstProp.cpp:
```
Attribute one = isa<FloatType>(elementType)
  ? rewriter.getFloatAttr(elementType, 1.0)
  : rewriter.getIntegerAttr(elementType, 1);
result = create.onnx.constant(DenseElementsAttr::get(resultType, one));
```